### ISSUE TITLE
A11y: Make filter map buttons actual buttons

### ISSUE THREAD

**Issue by bacitracin** [CodeForPhilly/vacant-lots-proj#337]
## Describe the task
The filter buttons  (priority level/parcel type/access process etc) are not reachable or operable via keyboard navigation or screenreader. This is because the filters are spans within divs, not actual buttons.

Changing them to buttons will fix this and will also make them visible in Windows High Contrast mode: 
<img width="309" alt="Screenshot 2024-02-10 at 3 11 25 PM" src="https://github.com/CodeForPhilly/vacant-lots-proj/assets/7329799/26087299-f950-4e85-8904-b228e4af8346">

## Acceptance Criteria

- [ ] You should be able to TAB to the filter buttons and press ENTER to turn them on or off

## Additional context
* Eventually we will want a new design for these filters. Showing that the filters are on/off via color breaks WCAG rules.
* https://www.smashingmagazine.com/2022/06/guide-windows-high-contrast-mode/
* How to simulate WHCM in Chrome Devtools - https://devtoolstips.org/tips/en/emulate-forced-colors/

**Comment by marvieqa** [comment]
Additional context relates to: #300

**Comment by marvieqa** [comment]
Additional Requirements:

- If we make the filter map buttons actual `<button>`,  use `aria-pressed` attribute so it can communicate the state for screen reader users.
- Or consider marking up these filters as a checkbox instead.

**Comment by bacitracin** [comment]
@marvieqa what do you think of adding a checkmark to the filter to indicate that it's selected?

![Image](https://github.com/CodeForPhilly/vacant-lots-proj/assets/7329799/34bfa8ea-4e01-45d3-8f9b-0148de5fcbc0)

**Comment by bacitracin** [comment]
PR: https://github.com/CodeForPhilly/vacant-lots-proj/pull/352
